aten/src/ATen/TensorUtils.h

```diff
@@ -62,6 +62,7 @@ void checkNumel(CheckedFrom c, const TensorGeometryArg& t, int64_t numel);
 void checkSameNumel(CheckedFrom c, const TensorGeometryArg& t1, const TensorGeometryArg& t2);
 void checkAllSameNumel(CheckedFrom c, ArrayRef<TensorArg> tensors);
 void checkScalarType(CheckedFrom c, const TensorArg& t, ScalarType s);
+void checkScalarTypes(CheckedFrom c, const TensorArg& t, std::initializer_list<ScalarType> l);
```

Comment:
I think we should prefer `at::ArrayRef<ScalarType>`. It has an (implicit) constructor that takes initializer lists so call sites do not need to change.